app/src/main/java/com/example/campusguide/directions/GetDirectionsDialogFragment.kt

```diff
@@ -0,0 +1,74 @@
+package com.example.campusguide.directions
+
+import android.app.AlertDialog
+import android.app.Dialog
+import android.os.Bundle
+import android.view.View
+import android.widget.EditText
+import androidx.fragment.app.DialogFragment
+import com.example.campusguide.R
+import java.lang.IllegalStateException
+
+/**
+ * An Android fragment that contains a dialog window prompting the user to
+ * enter their start and end location for getting directions.
+ */
+class GetDirectionsDialogFragment constructor(private val options: DirectionsDialogOptions) :
+    DialogFragment() {
+
+    /**
+     * Options for creating a Directions dialog.
+     * @param start Autofills the start location to the specified value
+     * @param end Autofills the end location to the specified value
+     * @param onConfirm Callback that will be executed when the user confirms the dialog window
+     */
+    class DirectionsDialogOptions constructor(
+        val start: String?, val end: String?,
+        val onConfirm: (String, String) -> Unit
+    )
+
+    /**
+     * Called when the dialog window is created.
+     */
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        // Creates a scope where the current activity is bound to the variable "it"
+        return activity?.let {
+
+            val inflater = requireActivity().layoutInflater
+            val view = inflater.inflate(R.layout.directions_dialog_layout, null)
+
+            setDefaultLocations(view)
+
+            val builder = AlertDialog.Builder(it)
+                .setView(view)
+                .setMessage("Enter start and end location")
+                .setPositiveButton("Go") { _, _ ->
+                    val startEditText =
+                        dialog?.findViewById<EditText>(R.id.startLocationTextInput)
+                    val endEditText = dialog?.findViewById<EditText>(R.id.endLocationTextInput)
+
+                    val start = startEditText?.text.toString()
+                    val end = endEditText?.text.toString()
+
+                    options.onConfirm(start, end)
+                }
+                .setNegativeButton("Cancel") { dialog, _ ->
+                    dialog.cancel()
+                }
+
+            return builder.create()
+        } ?: throw IllegalStateException("Activity cannot be null")
```

Comment:
is this supposed to be thrown when the start and/or end are left empty?

Reply:
Nope! The start and/or end can be left empty.
Although i got the code from a sample in the Android docs, I'm fairly certain this is thrown if you create the dialog window outside of an activity, so it crashes instead of silently doing nothing.

Reply:
Changed: Now if you leave the start or end blank, an error message will appear and no route will be displayed.